docs/reference/indices/flush.asciidoc

```diff
@@ -1,13 +1,16 @@
 [[indices-flush]]
 == Flush
 
-The flush API allows to flush one or more indices through an API. The
-flush process of an index makes sure that any data that is currently only
-persisted in the <<index-modules-translog,transaction log>> is also permanently
-persisted in Lucene. This reduces recovery times as that data doesn't need to be
-reindexed from the transaction logs after the Lucene indexed is opened. By
-default, Elasticsearch uses heuristics in order to automatically
-trigger flushes as required. It is rare for users to need to call the API directly.
+The flush process of an index makes sure that any data that is currently only
+stored in the <<index-modules-translog,transaction log>> is also permanently
+stored in Lucene. When restarting, {es} replays any unflushed operations from
+the transaction log into Lucene to ensure that searches do not return stale
```

Comment:
I liked the previous explanation better 👼 The real reason is that we want to bring lucene back to where it was before the shard was shutdown and lucene resets itself to the last time it was flushed. Stale searches are only one aspect of it.

Reply:
That's ok with me - see updated wording in https://github.com/elastic/elasticsearch/pull/46245/files#diff-31f3afe6b50f556a95a396324be535e5R6